src/internal/pfsdb/repos.go

```diff
@@ -380,3 +380,25 @@ func UpdateRepo(ctx context.Context, tx *pachsql.Tx, id RepoID, repo *pfs.RepoIn
 	}
 	return nil
 }
+
+// RepoExistsByName returns nil if the repo exists, or an error otherwise.
+func RepoExistsByName(ctx context.Context, tx *pachsql.Tx, repoProject, repoName, repoType string) error {
+	row := tx.QueryRowContext(ctx, `SELECT 1 FROM pfs.repos r LEFT JOIN core.projects p ON p.id = r.project_id `+
```

Comment:
This LEFT JOIN can just be a JOIN, since `repos.project_id` must reference an existing project.